### ISSUE TITLE
Idea: Keep count / identifiers of all Ref/Deferred in runtime

### ISSUE THREAD

**Issue by kubukoz** [typelevel/cats-effect#931]
This probably goes into CE3 territory, but it would be nice if we had more productionreadiness-friendly utilities in the runtime. It might be a bit too heavy in performance-sensitive code, but for most users having an extra reference for each Ref or Deferred would probably not be that much memory pressure.

Some extra ideas in the area:
- being able to access the identity of the fiber that spawned/changed a Ref/Deferred, the fibers waiting on a Deferred
- looking at the tree of fibers in the runtime, along with identities of fibers waiting on them / having canceled them

On top of such registries, we could probably build some useful debugging/monitoring tools, an example of something similar in the wild is https://github.com/ScalaConsultants/panopticon-tui.

**Comment by djspiewak** [comment]
Relates to some of the fiber identity ideas.
